tabsie/server.R

```diff
@@ -255,18 +257,24 @@ shinyServer(
     output$freqTable <- renderTable({#validation done before this is called, no need to repeat
       if (!valAuth) return;#break processing of not authorized.
       pdata = getpData(input$filter, serverDataDic, serverData)
-      addmargins(table(pdata[,c(input$xVal,input$yVal)]))
-    })
+      if(input$xVal==input$yVal) {
+        data.frame(Counts=cbind(summary(pdata[,input$xVal])))
+      } else {
+        as.data.frame.matrix(addmargins(table(pdata[,c(input$xVal,input$yVal)])))
```

Comment:
The `table()` command lets you create a convenient frequency table. However, `renderTable()` messes up the output and creates this long, skinny, hard-to-read table. Because a `table` object has a very different structure than a `data.frame`. So now we coerce it to a `data.frame` by using the `as.data.frame.matrix()` command I learned about just today.